txcache/autoClean_test.go

```diff
@@ -191,41 +135,50 @@ func TestTxCache_AutoClean_Dummy(t *testing.T) {
 	})
 
-	t.Run("with duplicated nonces", func(t *testing.T) {
-		boundsConfig := createMockTxBoundsConfig()
-		cache := newUnconstrainedCacheToTest(boundsConfig)
+}
 
-		session := txcachemocks.NewSelectionSessionMock()
-		session.SetNonce([]byte("alice"), 1)
+// helper function for creating a new unconstrained cache with a given size
+func newTxPoolWithN(size int, session *txcachemocks.SelectionSessionMock) *TxCache {
+	boundsConfig := createMockTxBoundsConfig()
+	cache := newUnconstrainedCacheToTest(boundsConfig)
+	for i := range size {
+		cache.AddTx(createTx([]byte(fmt.Sprintf("hash-%d", i)), fmt.Sprintf("sender-%d", i), uint64(i)))
+		session.SetNonce([]byte(fmt.Sprintf("sender-%d", i)), uint64(i-1))
+	}
+	return cache
+}
 
-		cache.AddTx(createTx([]byte("hash-alice-3a"), "alice", 3))
-		cache.AddTx(createTx([]byte("hash-alice-1"), "alice", 1))
-		cache.AddTx(createTx([]byte("hash-alice-2"), "alice", 2))
-		cache.AddTx(createTx([]byte("hash-alice-3b"), "alice", 3).withGasPrice(oneBillion * 2))
-		cache.AddTx(createTx([]byte("hash-alice-3c"), "alice", 3))
-		cache.AddTx(createTx([]byte("hash-alice-4"), "alice", 4))
-
-		// Check that the duplicates are removed
-		evicted := cache.Cleanup(session, 5, math.MaxInt, 1000*selectionLoopMaximumDuration)
-		require.Equal(t, uint64(2), evicted) // duplicates for nonce 3
-
-		// Check that the duplicates were removed based on their lower priority
-		listForAlice, _ := cache.txListBySender.getListForSender("alice")
-		require.Equal(t, 4, listForAlice.items.Len())
-
-		expected := map[uint64][]byte{
-			1: []byte("hash-alice-1"),
-			2: []byte("hash-alice-2"),
-			3: []byte("hash-alice-3b"),
-			4: []byte("hash-alice-4"),
-		}
-
-		for element := listForAlice.items.Front(); element != nil; element = element.Next() {
-			tx := element.Value.(*WrappedTransaction)
-			expectedHash, ok := expected[tx.Tx.GetNonce()]
-			require.True(t, ok, "Unexpected nonce %d", tx.Tx.GetNonce())
-			require.Equal(t, expectedHash, tx.TxHash)
-		}
+func BenchmarkAddressShuffling(b *testing.B) {
+	sizes := []int{1000, 10000, 50000, 100000}
+	for _, size := range sizes {
+		b.Run(fmt.Sprintf("size=%d", size), func(b *testing.B) {
+			// prepare pool
+			session := txcachemocks.NewSelectionSessionMock()
+			cache := newTxPoolWithN(size, session)
+
+			b.ResetTimer()
+			for i := 0; i < b.N; i++ {
+				senderAddresses := cache.txListBySender.backingMap.Keys()
+				shuffleSendersAddresses(senderAddresses, uint64(i))
+			}
+		})
+	}
+}
 
-	})
+func BenchmarkCleanup(b *testing.B) {
+	sizes := []int{1000, 10000, 50000, 100000}
+	for _, size := range sizes {
+		b.Run(fmt.Sprintf("size=%d", size), func(b *testing.B) {
+			b.ResetTimer()
+			for i := 0; i < b.N; i++ {
+				b.StopTimer()
+				session := txcachemocks.NewSelectionSessionMock()
+				cache := newTxPoolWithN(size, session)
+				b.StartTimer()
+
+				_ = cache.Cleanup(session, uint64(i), math.MaxInt, 1000*selectionLoopMaximumDuration)
```

Comment:
Can we use a different variable, other than `selectionLoopMaximumDuration`, since here we do cleanup? Simply hardcode a constant here.